src/System.Net.Sockets/src/System/Net/Sockets/UnixDomainSocketEndPoint.Unix.cs

```diff
@@ -70,7 +78,17 @@ internal UnixDomainSocketEndPoint(SocketAddress socketAddress)
                     _encodedPath[i] = socketAddress[s_nativePathOffset + i];
                 }
 
-                _path = s_pathEncoding.GetString(_encodedPath, 0, _encodedPath.Length);
+                // Strip trailing null of pathname socket addresses.
+                int length = _encodedPath.Length;
+                bool isAbstract = _encodedPath[0] == 0;
```

Comment:
Can we be defensive here?  While it shouldn't happen that _encodedPath.Length == 0, there's nothing in this method that technically prevents that from happening.

Reply:
Oh, nevermind, I missed the `if (socketAddress.Size > s_nativePathOffset)` check earlier.